### PR TITLE
Move e2e-metal-ipi to run on the new baremetal lab

### PR DESCRIPTION
**Pull request by derekhiggins**
Add support for real baremetal jobs
switching the e2e-metal-ipi periodic job to use it

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh [hunk 1]
```diff
@@ -76,21 +76,92 @@ cat > packet-config.yaml <<-EOF
...
+    else
+        return 1
+    fi
+}
+
+CIRTYPE=cihost
+if [[ "$CLUSTERTYPE" == "baremetal" ]] ; then
+    CIRTYPE=cicluster
+    if [ -e "${CLUSTER_PROFILE_DIR}/ofcir_laburl" ] ; then
+        getCIR "$(cat ${CLUSTER_PROFILE_DIR}/ofcir_laburl)"
```
Comment (elfosardo) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:154]: can we use $OFCIRURL here?

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh [hunk 2]
```diff
@@ -76,21 +76,92 @@ cat > packet-config.yaml <<-EOF
...
         exit_with_success
     fi
+    exit_with_failure
+fi
+
+if [ -e "${CLUSTER_PROFILE_DIR}/ofcir_laburl" ] ; then
+    getCIR "$(cat ${CLUSTER_PROFILE_DIR}/ofcir_laburl)" && exit_with_success
```
Comment (elfosardo) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:161]: ditto
maybe remove L163-165 if we do

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh (hunk 2 above)
Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:161]: I'm not sure what you mean here, OFCIRURL doesn't exist at this point,
These lines check a different OFCIR server when compared to the 3 below, I'll add some comments to
explain whats going on

Comment (elfosardo) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:171]: nvm, I completely misread the script

**Review by elfosardo — APPROVED** [review]

**Comment by derekhiggins** [comment]
@wking @stbenjam this is a new baremetal job we're hoping to use to eventually replace some of the virtualised jobs, if you have a chance can you take a look.

**Comment by stbenjam** [comment]
/hold

Looks like step-registry-shellcheck has some complaints

**Comment by derekhiggins** [comment]
> /hold
> 
> Looks like step-registry-shellcheck has some complaints

I thought it was a pre-existing problem as I didn't change the line it was complaining about

I've disabled the test on that line as I want the env variable to be expanded client side

**Pull request merged**